java/trace/src/main/scala/org/typelevel/otel4s/java/trace/TraceScope.scala

```diff
@@ -47,67 +45,70 @@ private[java] object TraceScope {
     case object Noop extends Scope
   }
 
-  def fromIOLocal[F[_]: LiftIO: Sync](
+  private[java] def fromLocal[F[_]: Sync: Local[*[_], Scope]](
       default: JContext
-  ): F[TraceScope[F]] = {
+  ): TraceScope[F] = {
     val scopeRoot = Scope.Root(default)
 
-    IOLocal[Scope](scopeRoot).to[F].map { local =>
-      new TraceScope[F] {
-        val root: F[Scope.Root] =
-          Sync[F].pure(scopeRoot)
+    new TraceScope[F] {
+      val root: F[Scope.Root] =
+        Sync[F].pure(scopeRoot)
 
-        def current: F[Scope] =
-          local.get.to[F]
+      def current: F[Scope] =
+        Local[F, Scope].ask[Scope]
 
-        def makeScope(span: JSpan): Resource[F, Unit] =
-          for {
-            current <- Resource.eval(current)
-            _ <- createScope(nextScope(current, span))
-          } yield ()
+      def makeScope(span: JSpan): Resource[F, Unit] =
+        for {
+          current <- Resource.eval(current)
+          _ <- createScope(nextScope(current, span))
+        } yield ()
 
-        def rootScope: Resource[F, Unit] =
-          Resource.eval(current).flatMap {
-            case Scope.Root(_) =>
-              createScope(scopeRoot)
+      def rootScope: Resource[F, Unit] =
+        Resource.eval(current).flatMap {
+          case Scope.Root(_) =>
+            createScope(scopeRoot)
 
-            case Scope.Span(_, _, _) =>
-              createScope(scopeRoot)
+          case Scope.Span(_, _, _) =>
+            createScope(scopeRoot)
 
-            case Scope.Noop =>
-              createScope(Scope.Noop)
-          }
+          case Scope.Noop =>
+            createScope(Scope.Noop)
+        }
 
-        def noopScope: Resource[F, Unit] =
-          createScope(Scope.Noop)
+      def noopScope: Resource[F, Unit] =
+        createScope(Scope.Noop)
 
-        private def createScope(scope: Scope): Resource[F, Unit] =
+      private def createScope(scope: Scope): Resource[F, Unit] = {
+        val _ = scope
+        // There's no getAndSet.  This is inherently stateful.
+        /*
           Resource
             .make(local.getAndSet(scope).to[F])(p => local.set(p).to[F])
             .void
+         */
```

Comment:
And this here is where the wheels fall off.  Do we want tracing of more than `IOLocal`, or do we want `acquire`, `use`, and `release` to be siblings?  This was a matter of long debate in Natchez.

Reply:
Exposing the `Resource` as `Client#run` really shot us in the foot for this use case.  We used to accept the callback directly -- I think it was called `fetch`.  There's not a whole lot we can do with local semantics:

1. We can span whatever calls `Client#run`, but `Client#run` can't span it locally because it can't see it.
2. We could have a `Resource[F, A] => Resource[F, (F ~> F, A)]`, and the user of the resource would be responsible for calling the natural transformation for scoping.  But that's roughly as intrusive as tracing the caller of `Client#run` and more confusing.
3. We could create a new resource-like algebra with an `Intercept`, constructable from `Resource`.  But, gosh, that's disruptive.
4. Someone smarter than me could figure out how to shoehorn it into the existing constructors.  I more readily believe someone is smarter than me than that this is possible.